.travis.yml

```diff
@@ -56,26 +73,29 @@ install:
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m pip install --upgrade pytest flake8 "sphinx<3" sphinx_rtd_theme coverage codecov hypothesis attrs; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m pip install --upgrade mypy || true; fi
-script:
-  #windows build
-  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then source .travis/runPycairo.sh; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py sdist; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py bdist; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py install --root=_root; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py install --root="$(pwd)"/_root_abs; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py bdist_wheel; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python setup.py install --root=_root_setup; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then if [[ "${TRAVIS_PYTHON_VERSION:0:4}" != "pypy" ]] ; then python -m pip install .; fi; fi
-  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then python -m sphinx -W -a -E -b html -n docs docs/_build; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then curl -sS -L https://aka.ms/nugetclidl -o $TMP/nuget.exe; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then curl -sS -L https://github.com/preshing/cairo-windows/releases/download/$CAIRO_VERSION/cairo-windows-$CAIRO_VERSION.zip -o $TMP/cairo-windows-$CAIRO_VERSION.zip; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then 7z x $TMP/cairo-windows-$CAIRO_VERSION.zip -o$TMP; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then $TMP/nuget install python -Version $PYTHON_VERSION -OutputDirectory $TMP/nuget-$ARCH; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then export INCLUDE="$TMP/cairo-windows-$CAIRO_VERSION/include"; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then export LIB=$TMP/cairo-windows-$CAIRO_VERSION/lib/$ARCH; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then export PATH=$TMP/nuget-$ARCH/python.$PYTHON_VERSION/tools/:PATH=$TMP/nuget-$ARCH/python.$PYTHON_VERSION/tools/Scripts:$PATH; fi
+  - if [ "$TRAVIS_OS_NAME" == "windows" ]; then python -m pip install --upgrade wheel; fi
+  - python -m pip install --upgrade setuptools
+  - python -m pip install --upgrade pytest flake8 "sphinx<3" sphinx_rtd_theme coverage codecov hypothesis attrs
+  - python -m pip install --upgrade mypy || true
+  - if [[ "$TRAVIS_OS_NAME" != "osx" ]] && [[ "$TRAVIS_PYTHON_VERSION" != "3.8" ]] && [[ "${TRAVIS_PYTHON_VERSION:0:4}" != "pypy" ]]; then python -m pip install --upgrade pygame; fi
 
-deploy:
-  if: TRAVIS_OS_NAME = windows
-  provider: releases
-  file_glob: true
-  api_key: $GITHUBOAUTHTOKEN
-  file: dist/*
-  skip_cleanup: true
-  draft: true
+script:
+  - if [[ "$TRAVIS_OS_NAME" == "windows" ]] ; then cp $TMP/cairo-windows-$CAIRO_VERSION/lib/$ARCH/cairo.dll cairo/cairo.dll; fi
+  - python -m coverage run --branch setup.py test
+  - python -m codecov --required || true
```

Comment:
What does the above two lines do? I really don't understand.

Reply:
(handwaves) - checking how much coverage there is, I think it uploads to a service, where you can view it.

I haven't used it to much, the coverage bits where from newest pycairo master, not my own changes.

Reply:
So.. .discovered some things.   There is [a bug](https://github.com/chocolatey/choco/issues/1843) in current chocolatey (fixed in next version) that prevents this kind of search.

But in the meantime it's possible to downgrade chocolatey:

`choco install chocolatey -y --version 0.10.13 --allow-downgrade`

Then we can search all versions:

```shell
$ choco search -e python3 -a
Chocolatey v0.10.13
python3 3.9.0-rc1 [Approved]
python3 3.9.0-b5 [Approved]
python3 3.9.0-b4 [Approved]
[snipped a lot of versions]
```

Use grep to search for python 3.7 and remove the [Approved] column:
```shell
$ choco search -e python3 -a | grep 'python3 3.7' | head | cut -d' ' -f1,2
python3 3.7.9
python3 3.7.8
python3 3.7.7
python3 3.7.6.20200110
python3 3.7.6
python3 3.7.5
python3 3.7.4
python3 3.7.3
python3 3.7.2
python3 3.7.2-rc1
```

^ It looks like they come back in the order they were added, so which seems like what we want.

Get exact python version given "3.7":
```
choco search -e python3 -a | grep 'python3 3.7' | head | cut -d' ' -f2 | head -1
3.7.9
```